httpx/utils.py

```diff
@@ -94,8 +95,9 @@ def guess_json_utf(data: bytes) -> typing.Optional[str]:
 NETRC_STATIC_FILES = (Path("~/.netrc"), Path("~/_netrc"))
 
 
-def get_netrc_login(host: str) -> typing.Optional[typing.Tuple[str, str, str]]:
-    NETRC_FILES = (Path(os.getenv("NETRC", "")),) + NETRC_STATIC_FILES
+@functools.lru_cache(1)
```

Comment:
More of a general question about the approach to caching. If the *contents* of the netrc file change wouldn't we be returning stale crendentials?

I guess it's not very common but it could be quite confusing for the user.

Reply:
:+1: this change probably needs to be mentioned in the documentation

Reply:
Totally agree we need to document this within the netrc authentication section. :) Then this looks good to merge for me.

Reply:
".netrc file is cached after first use by env file (not by host)"

Reply:
Should we mentione about reset the cache if the file content is changed

Reply:
We should mention that once HTTPX has been loaded that content within the `netrc` file is only checked once. Recommended way to refresh the cache would be to restart the interpreter.

Reply:
Behaviorally the *ideal* would be to have the netrc file loaded and cached on the Client instance, once and once only. (Either at instantiation, or one the first request.)

LRU caching on the host lookup, is an okay light-touch approach tho.